### PR TITLE
Fixes and namespace selection support on build steps

### PR DESCRIPTION
**Pull request by gsanchezu**
- Fix JENKINS-37176 The labels field is not set
- Fix YAML validation when Cloud has not been saved
- Add namespace selection support to build steps

@DiegoSanjuan, ACK? 👍

### CODE REVIEW

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/ebx_pull_request_jenkins_plugin_kubernetes/37/ for more details.

File: src/main/java/com/elasticbox/jenkins/k8s/plugin/builders/ChartBuildStepDescriptor.java [hunk 1]
```diff
@@ -10,6 +10,9 @@
 
 import com.google.inject.Injector;
 
+import com.elasticbox.jenkins.k8s.plugin.clouds.KubernetesCloudParams;
```
Comment (DiegoSanjuan) [inline src/main/java/com/elasticbox/jenkins/k8s/plugin/builders/ChartBuildStepDescriptor.java:13]: should we have all the k8s imports together and in alpha order, shouldn't we?

File: src/main/java/com/elasticbox/jenkins/k8s/plugin/builders/DeleteChartBuildStep.java [hunk 2]
```diff
@@ -13,8 +13,8 @@
 
 import com.elasticbox.jenkins.k8s.util.TaskLogger;
 import com.elasticbox.jenkins.k8s.chart.ChartRepo;
```
Comment (DiegoSanjuan) [inline src/main/java/com/elasticbox/jenkins/k8s/plugin/builders/DeleteChartBuildStep.java:15]: same, order...

File: src/main/java/com/elasticbox/jenkins/k8s/plugin/builders/DeployChartBuildStep.java [hunk 3]
```diff
@@ -14,8 +14,8 @@
 import com.elasticbox.jenkins.k8s.util.TaskLogger;
 import com.elasticbox.jenkins.k8s.chart.Chart;
 import com.elasticbox.jenkins.k8s.chart.ChartRepo;
-import com.elasticbox.jenkins.k8s.plugin.clouds.KubernetesCloud;
 import com.elasticbox.jenkins.k8s.repositories.ChartRepository;
```
Comment (DiegoSanjuan) [inline src/main/java/com/elasticbox/jenkins/k8s/plugin/builders/DeployChartBuildStep.java:17]: same

**Comment by DiegoSanjuan** [comment]
ACK, just minor comments about the imports

**Comment by osanjuan** [comment]
hi Guillermo, can you address the comments please?

thanks!

**Comment by allyourbotsarebelongtous** [comment]
The following instances are being terminated: https://elasticbox.com/services/instances/i-74stc1

File: src/main/java/com/elasticbox/jenkins/k8s/plugin/builders/DeleteChartBuildStep.java (hunk 2 above)
Comment (gsanchezu) [inline src/main/java/com/elasticbox/jenkins/k8s/plugin/builders/DeleteChartBuildStep.java:15]: 👍 done!

File: src/main/java/com/elasticbox/jenkins/k8s/plugin/builders/DeployChartBuildStep.java (hunk 3 above)
Comment (gsanchezu) [inline src/main/java/com/elasticbox/jenkins/k8s/plugin/builders/DeployChartBuildStep.java:17]: 👍 done!

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/ebx_pull_request_jenkins_plugin_kubernetes/38/ for more details.

**Comment by gsanchezu** [comment]
Jenkins test this please again! 👍

**Comment by allyourbotsarebelongtous** [comment]
Deployed ElasticKube instance is available at http://10.0.51.236 and Jenkins instance at https://10.0.51.245

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/ebx_pull_request_jenkins_plugin_kubernetes/39/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
The following instances are being terminated: https://ops.elasticbox.com/services/instances/i-6stcku, https://elasticbox.com/services/instances/i-336gpm

**Pull request merged**